pyproject.toml

```diff
@@ -253,68 +256,17 @@ omit = [
 [tool.setuptools_scm]
 write_to = "jwst/_version.py"
 
-[tool.ruff]
-exclude = [
-    "jwst/extern",
-    "docs",
-    ".tox",
-    ".eggs",
-    "build",
-    "*.ipynb",
-]
-line-length = 130
-
-[tool.ruff.lint]
-select = [
-    "F", # Pyflakes (part of default flake8)
-    "E", # pycodestyle (part of default flake8)
-    "W", # pycodestyle (part of default flake8)
-    # "A", # flake8-builtins (prevent shadowing of builtins)
-    # "ANN",    # flake8-annotations (best practices for type annotations)
-    # "BLE",    # flake8-blind-except (prevent blind except statements)
-    # "C4",     # flake8-comprehensions (best practices for comprehensions)
-    # "T10",    # flake8-debugger (prevent debugger statements in code)
-    # "FA",     # flake8-future-annotations (correct usage future annotations)
-    # "ISC",    # flake8-implicit-str-concat (prevent implicit string concat)
-    # "ICN",    # flake8-import-conventions (enforce import conventions)
-    # "INP",    # flake8-no-pep420 (prevent use of PEP420, i.e. implicit name spaces)
-    # "RSE",    # flake8-raise (best practices for raising exceptions)
-    # "SLOT",   # flake8-slots (require __slots__ for immutable classes)
-    # "TID",    # flake8-tidy-imports (prevent banned api and best import practices)
-    # "TCH",    # flake8-type-checking (move type checking imports into type checking blocks)
-    # "INT",    # flake8-gettext (when to use printf style strings)
-    # "PGH",    # pygrep (simple grep checks)
-    # "FLY",    # flynt (f-string conversion where possible)
-    # "LOG",
-]
-ignore = [
-    "E741", # ambiguous variable name (O/0, l/I, etc.)
-]
-
-[tool.ruff.lint.pydocstyle]
-convention = "numpy"
-
-[tool.ruff.lint.flake8-annotations]
-ignore-fully-untyped = true  # Turn of annotation checking for fully untyped code
-
-[tool.ruff.lint.extend-per-file-ignores]
-"jwst/**/__init__.py" = [
-    "E402", # module-level import not at top of file
-    "F401", # unused import
-    "F403", # import with *
-    "F405", # ambiguous import from *
-]
-"jwst/fits_generator/*" = [
-    "F821", # undefined name
-]
-"jwst/fits_generator/tests/*" = [
-    "F841", # unused variable
-]
-"jwst/associations/tests*" = [
-    "F841", # unused variable
-]
-"jwst/ramp_fitting/tests/*" = [
-    "E402", # module-level import not at top of file
+[tool.numpydoc_validation]
+checks = [
+    "all",
+    "EX01", # No examples section found
+    "SA01", # See Also section not found
+    "ES01", # No extended summary found
+    "GL08", # Object does not have a docstring. Ruff catches these, and allows more granular ignores.
+    "PR01", # Parameters not documented. Already caught by ruff.
+    "PR09", # Parameter description should finish with a period
+    "RT02", # First line of return should contain only the type
+    "RT05", # Return value description should finish with a period
 ]
 
 [tool.mypy]
```

Comment:
Are the `mypy` sections are useful if someone elects to install and run `mypy` separately from the listed requirements, or can these sections be removed?

Reply:
Also, I'm pretty sure the mypy CI check also reads from here